eland/ml/pytorch/transformers.py

```diff
@@ -573,7 +573,13 @@ def _prepare_inputs(self) -> transformers.BatchEncoding:
 
 
 class TransformerModel:
-    def __init__(self, model_id: str, task_type: str, quantize: bool = False):
+    def __init__(
+        self,
+        model_id: str,
+        task_type: str,
+        es_version: Tuple[int, int, int],
```

Comment:
I'm a little apprehensive about forcing users to pass an `es_version` directly into this class, is there any other way we can handle this? (Should we wait to do any version-specific logic until the model needs to be serialized to Elasticsearch, at which point we'll have a client instance?)

Reply:
The `TransformerModel` class is responsible for loading the model and converting it to a format that can be uploaded to Elasticsearch. The `PyTorchModel` class does the actual upload and has the client. Yes is would be nice to not require the version here but it would require a significant refactoring of those 2 classes that would probably break existing usages.

The version is required so that new features can be supported and to avoid creating incompatible configurations on older clusters, I expect we will see more of this type of logic in the future.

I've made the version parameter optional so that upgrade won't break any existing scripts using the class. If the version is not set only the most basic settings are used. I've also added docs explaining the usage. 

Another option is to pass the client rather than the cluster version to the `TransformerModel` class but then you need a working client and cluster to get the version when the use case might be just to save the model locally.

Reply:
Sounds good, if you're happy with the implementation then I am okay with it.